cosmotech/data_update_quest/core/database/redis/client.py

```diff
@@ -21,16 +21,22 @@ def get_redis_client(host, port, password):
 
 
 def get_redis_indexes(r, index_list: Optional[list[str]]):
-    if not index_list:
-        index_list = r.execute_command("FT._LIST")
 
     LOGGER.info(T("data_update_quest.core.redis_dump.redis_index"))
+
     indexes = {}
+    if index_list:
```

Comment:
Instead of duplicating the for loop in the to part of `if else` you could keep it at the end and create a `full_index_list` with the following code to iterate on it :
```python
if index_list:
    full_index_list = list(f"com.cosmotech.{index_name.lower()}.domain.{index_name.capitalize()}Idx" for index_name in index_list)
else:
    full_index_list = r.execute_command("FT._LIST")
```